### PR TITLE
[MAINT] deal with new warnings related to sklearn 1.3

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to #3800 

Changes proposed in this pull request:

silence some warnings in doc build:
- setting defaults for dual explicitly in LinearSVC
- settings default in  when fitting sklearn.linear_model.OrthogonalMatchingPursuit

### CODE REVIEW

File: examples/02_decoding/plot_miyawaki_encoding.py [hunk 1]
```diff
@@ -143,11 +143,9 @@
 for train, test in cv.split(X=stimuli):
     # we train the Ridge estimator on the training set
     # and predict the fMRI activity for the test set
-    predictions = (
-        Ridge(alpha=100.0)
-        .fit(stimuli.reshape(-1, 100)[train], fmri_data[train])
-        .predict(stimuli.reshape(-1, 100)[test])
-    )
+    predictions = estimator.fit(
```
Comment (Remi-Gau) [inline examples/02_decoding/plot_miyawaki_encoding.py:146]: `estimator` is defined above but was not used in this file

File: nilearn/decoding/decoder.py [hunk 2]
```diff
@@ -52,8 +52,8 @@
 
 SUPPORTED_ESTIMATORS = dict(
     svc_l1=LinearSVC(penalty="l1", dual=False, max_iter=10000),
-    svc_l2=LinearSVC(penalty="l2", max_iter=10000),
-    svc=LinearSVC(penalty="l2", max_iter=10000),
+    svc_l2=LinearSVC(penalty="l2", dual=True, max_iter=10000),
```
Comment (Remi-Gau) [inline nilearn/decoding/decoder.py:55]: This just sets the default value to silence the warning.

**Review by ymzayek — COMMENTED** [review]
If relevant, we can probably also make the changes to unit tests in this PR. For example the warning for the `dual` default shows up in `nilearn/decoding/tests/test_searchlight.py`, see https://github.com/nilearn/nilearn/actions/runs/5454471723/jobs/9924639057?pr=3801#step:8:5327

**Review by bthirion — COMMENTED** [review]
LGTM so far

File: nilearn/decomposition/dict_learning.py [hunk 3]
```diff
@@ -291,15 +291,16 @@ def _raw_fit(self, data):
 
         dict_init = self.loadings_init_
 
-        n_iter = ((n_features - 1) // self.batch_size + 1) * self.n_epochs
+        max_iter = ((n_features - 1) // self.batch_size + 1) * self.n_epochs
 
         if self.verbose:
             print("[DictLearning] Learning dictionary")
+        # TODO: turn n_iter to max_iter when dropping python 3.7
```
Comment (Remi-Gau) [inline nilearn/decomposition/dict_learning.py:298]: probably worth opening an issue to track some of the things we may want to check when dropping python 3.7

File: examples/02_decoding/plot_miyawaki_encoding.py
Comment (ymzayek) [inline examples/02_decoding/plot_miyawaki_encoding.py:1]: Also remove `normalize=False,` in https://github.com/nilearn/nilearn/blob/cd76cfc50d040fb77cf939fe709de625a7bbbd79/examples/02_decoding/plot_miyawaki_encoding.py#L236-L238 ?

File: examples/02_decoding/plot_miyawaki_reconstruction.py [hunk 4]
```diff
@@ -160,7 +160,7 @@ def flatten(list_of_2d_array):
         [
             ("selection", SelectKBest(f_classif, k=500)),
             ("scl", StandardScaler()),
-            ("clf", OMP(normalize=False, n_nonzero_coefs=10)),
+            ("clf", OMP(normalize="deprecated", n_nonzero_coefs=10)),
```
Comment (ymzayek) [inline examples/02_decoding/plot_miyawaki_reconstruction.py:163]: normalize parameter can just be deleted no?

**Review by ymzayek — COMMENTED** [review]
LGTM otherwise!

File: nilearn/decomposition/dict_learning.py (hunk 3 above)
Comment (ymzayek) [reply nilearn/decomposition/dict_learning.py:298]: It doesn't hurt, but we can also open a new milestone for release 0.11 and document there because that will be when we drop 3.7

File: examples/02_decoding/plot_miyawaki_encoding.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_encoding.py:1]: yup that works: checked and no warning thrown

File: examples/02_decoding/plot_miyawaki_reconstruction.py (hunk 4 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py]: yup that works: checked and no warning thrown

**Review by ymzayek — APPROVED** [review]

**Comment by Remi-Gau** [comment]
thanks for the review @ymzayek
